core/editable.js

```diff
@@ -1563,6 +1567,16 @@
 			// Create a new walker.
 			var walker = new CKEDITOR.dom.walker( walkerRng );
 
+			// Check if range include more than one list.
+			var multipleList = checkForMultipleList( range );
+
+			if ( multipleList ) {
+				return {
```

Comment:
I'm wondering if we could separate somehow verifying if there are multiple lists in the range from the old code? For example, extract a separate method for that? Returning an object with a flag inside seems to be confusing.